client/jetpack-connect/authorize.js

```diff
@@ -881,26 +884,41 @@ export class JetpackAuthorize extends Component {
 		// 'jetpack/magic-link-signup' feature flag) are created with a username based on the user's
 		// email address. For this reason, we want to display both the username and the email address
 		// so users can start making the connection between the two immediately. Otherwise, users might
-		// not recognize their username since they didn't created it.
+		// not recognize their username since they didn't create it.
 
-		// translators: %(user) is user's Display Name (Eg Connecting as John Doe) and %(email) is the user's email address
-		let text = translate(
-			'Connecting as {{strong}}%(user)s{{/strong}} ({{strong}}%(email)s{{/strong}})',
-			{
-				args: { email: this.props.user.email, user: this.props.user.display_name },
-				components: { strong: <strong /> },
-			}
-		);
+		// Note: We make an exception for Woo onboardings, since in these cases the creation of a Jetpack account
+		// is an intermediate step and the user will be redirected to the WooCommerce onboarding flow.
+		// Seeing this new username/email address can cause confusion because they have already set up
+		// a Woo account under their own email address.
 
-		if ( authorizeSuccess || this.props.isAlreadyOnSitesList ) {
+		let text = '';
+
+		if ( isWooOnboarding && isJetpackMagicLinkSignUpFlow ) {
+			text = translate( 'Connecting your account' );
+
+			if ( authorizeSuccess || this.props.isAlreadyOnSitesList ) {
+				text = translate( 'Account connected successfully' );
+			}
+		} else {
 			// translators: %(user) is user's Display Name (Eg Connecting as John Doe) and %(email) is the user's email address
 			text = translate(
-				'Connected as {{strong}}%(user)s{{/strong}} ({{strong}}%(email)s{{/strong}})',
+				'Connecting as {{strong}}%(user)s{{/strong}} ({{strong}}%(email)s{{/strong}})',
 				{
 					args: { email: this.props.user.email, user: this.props.user.display_name },
 					components: { strong: <strong /> },
 				}
 			);
+
+			if ( authorizeSuccess || this.props.isAlreadyOnSitesList ) {
+				// translators: %(user) is user's Display Name (Eg Connecting as John Doe) and %(email) is the user's email address
+				text = translate(
+					'Connected as {{strong}}%(user)s{{/strong}} ({{strong}}%(email)s{{/strong}})',
+					{
+						args: { email: this.props.user.email, user: this.props.user.display_name },
+						components: { strong: <strong /> },
+					}
+				);
+			}
```

Comment:
Minor: It would be great if we can simplify this block of code like the one below to make it easier to understand and maintain. 🙂 


```js
if ( isWooOnboarding && isJetpackMagicLinkSignUpFlow ) {
	text =
		authorizeSuccess || this.props.isAlreadyOnSitesList
			? translate( 'Account connected successfully' )
			: translate( 'Connecting your account' );
} else {
	const connectionStatus =
		authorizeSuccess || this.props.isAlreadyOnSitesList ? 'Connected' : 'Connecting';

	// translators: %(connectionStatus) is the connection status (Eg Connecting) and %(user) is user's Display Name (Eg John Doe) and %(email) is the user's email address
	text = translate(
		'%(connectionStatus)s as {{strong}}%(user)s{{/strong}} ({{strong}}%(email)s{{/strong}})',
		{
			args: {
				connectionStatus,
				email: this.props.user.email,
				user: this.props.user.display_name,
			},
			components: { strong: <strong /> },
		}
	);
}
```


or even better by returning the text directly:

```js
  if ( isWooOnboarding && isJetpackMagicLinkSignUpFlow ) {
    return authorizeSuccess || this.props.isAlreadyOnSitesList
      ? translate( 'Account connected successfully' )
      : translate( 'Connecting your account' );
  }

  const connectionStatus =
    authorizeSuccess || this.props.isAlreadyOnSitesList ? 'Connected' : 'Connecting';

  // translators: %(connectionStatus) is the connection status (Eg Connecting) and %(user) is user's Display Name (Eg John Doe) and %(email) is the user's email address
  return translate(
    '%(connectionStatus)s as {{strong}}%(user)s{{/strong}} ({{strong}}%(email)s{{/strong}})',
    {
      args: {
        connectionStatus,
        email: this.props.user.email,
        user: this.props.user.display_name,
      },
      components: { strong: <strong /> },
    }
  );
```

Reply:
Great suggestion. Implemented in ab4dd3c8d4.